crates/iota-types/src/mock_checkpoint_builder.rs

```diff
@@ -68,6 +68,24 @@ impl MockCheckpointBuilder {
             .push(VerifiedExecutionData::new(transaction, effects))
     }
 
+    /// Override the next checkpoint number to generate.
+    /// This can be useful to generate checkpoints with specific sequence
+    /// numbers.
+    pub fn override_next_checkpoint_number(
+        &mut self,
+        checkpoint_number: u64,
+        validator_keys: &impl ValidatorKeypairProvider,
+    ) {
+        if checkpoint_number > 0 {
+            let mut summary = self.previous_checkpoint.as_ref().unwrap().data().clone();
```

Comment:
What if the former `previous_checkpoint` was already None?
Maybe an expect would be better here if we can't avoid the panic (for tests only anyways).
Can we please still add a comment here.... it's a bit hard to understand why the function is called "overwrite next checkpoint number", but then you do some magic here with the previous checkpoint.